src/core/toga/evented.py

```diff
@@ -0,0 +1,164 @@
+"""toga.evented - Base classes for objects that expose events
+
+This module defines classes to be used for creating objects that need to expose
+events in a way that allows other objects to set callback to be called when
+event occur. An example for this is Widgets that invoke callbacks when users
+interact with them.
+
+Here is an example of how to use the classes that can be found here.
+
+    >>> class AClickableWidget(Evented):
+    ...     on_click = Event('Called when wodget is clicked')
+    ...
+    ...     def __init__(self, **kwargs):
+    ...         super().__init__(**kwargs)
+    ...         # This is typically done via the `factory` argument
+    ...         self._impl = AClickableWidgetBackend(interface=self)
+    ...
+    >>> class AClickableWidgetBackend:
+    ...     def __init__(self, interface):
+    ...         self.interface = interface
+    ...
+    ...     def emulate_click(self, which_button):
+    ...         self.interface.raise_event('on_click', which_button)
+    ...
+    >>> def click_handler(widget, which_button):
+    ...     print(f"I was clicked with {which_button} mouse button!")
+    ...
+    >>> widget = AClickableWidget(on_click=click_handler)
+    >>> widget._impl.emulate_click("Left")
+    I was clicked with Left mouse button!
+    >>> widget._impl.emulate_click("Right")
+    I was clicked with Right mouse button!
+
+Note: Typically only top-level classes such as `Widget` or `Window` should
+inherit directly from `Evented`, others should inherit from those classes
+instead.
+"""
+from toga.handlers import wrapped_handler
+
+
+class Evented:
```

Comment:
This is a bit of a bikeshed, but following the naming themes in Toga generally, I'd suggest this should be `EventSource`, rather than `Evented`, as class names are nominative, rather than adjectival.

Reply:
ok, will fix.